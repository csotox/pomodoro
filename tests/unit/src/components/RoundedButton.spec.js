import { shallowMount } from '@vue/test-utils';
import RoundedButton from '@/components/RoundedButton.vue';

function createRoundedButtonView() {
  return shallowMount(RoundedButton, {
    slots: {
      default: 'Haz tu primer pomodoro',
    },
  });
}

describe('RoundedButton.vue', () => {
  it('should render properly', () => {
    const roundedButton = createRoundedButtonView();
    expect(roundedButton).toMatchSnapshot();
  });

  describe('Methods', () => {
    describe('onClick', () => {
      it('should emit an event named click', () => {
        const roundedButton = createRoundedButtonView();

        roundedButton.find('.rounded-button').trigger('click');

        expect(roundedButton.emitted('click')).toBeTruthy();
      });
    });
  });
});
