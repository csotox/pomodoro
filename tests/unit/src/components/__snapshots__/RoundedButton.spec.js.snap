// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton.vue should render properly 1`] = `<button class="rounded-button">Haz tu primer pomodoro</button>`;
